service/client_state.cc

```diff
@@ -53,13 +53,64 @@
 #include "db/system_distributed_keyspace.hh"
 #include "database.hh"
 #include "cdc/log.hh"
+#include "concrete_types.hh"
 
 thread_local api::timestamp_type service::client_state::_last_timestamp_micros = 0;
 
 void service::client_state::set_login(auth::authenticated_user user) {
     _user = std::move(user);
 }
 
+service::client_state::client_state(external_tag, auth::service& auth_service, timeout_config timeout_config, const socket_address& remote_address, bool thrift)
+        : _is_internal(false)
+        , _is_thrift(thrift)
+        , _remote_address(remote_address)
+        , _auth_service(&auth_service)
+        , _default_timeout_config(timeout_config)
+        , _timeout_config(timeout_config) {
+    if (!auth_service.underlying_authenticator().require_authentication()) {
+        _user = auth::authenticated_user();
+    }
+}
+
+future<> service::client_state::update_per_role_params() {
+    if (!_user || auth::is_anonymous(*_user)) {
+        return make_ready_future<>();
+    }
+    //FIXME: replace with a coroutine once they're widely accepted
```

Comment:
```console
$ git grep -E 'co_await|co_return' | wc -l
44
```

Reply:
Yes, but
```bash
$ git grep -E 'co_await|co_return' | grep -c -v raft
0
```
, and raft has conditional compilation. I still expect frowns when pushing coroutines to unconditionally compiled code.